### PR TITLE
feat: Notify for VS restore on uno.sdk change

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/17495

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

Provide a banner to restart VS in place when the uno.sdk has changed.

### CODE REVIEW

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.3

File: src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+		{
+			Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
+
+			var globalJsonPath = GetGlobalJsonPath();
+
+			ObserveChanges(globalJsonPath);
+		}
+		catch (Exception e)
+		{
+			_debugAction($"GlobalJsonObserver: Failed to start " + e);
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs:56]:
```suggestion
			_debugAction($"GlobalJsonObserver: Failed to start - {e.Message}");
```

File: src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+				try
+				{
+					var newVersion = ReadUnoSdkVersion(globalJsonPath);
+
+					if (newVersion != unoSdkVersion)
+					{
+						unoSdkVersion = newVersion;
+
+						// Reload all projects that use the Uno.SDK
+						_debugAction($"GlobalJsonObserver: Uno.Sdk version changed to {unoSdkVersion}.");
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs:84]:
```suggestion
						_debugAction($"GlobalJsonObserver: Uno.Sdk version changed to {unoSdkVersion}");
```

File: src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+							if (shell is IVsShell4 shell4)
+							{
+								((IVsShell3)shell).IsRunningElevated(out var elevated);
+								var type = elevated ? __VSRESTARTTYPE.RESTART_Elevated : __VSRESTARTTYPE.RESTART_Normal;
+								var hr = shell4.Restart((uint)type);
+							}
+						}
+						else if (e.ActionItem == moreInformationVSItem)
+						{
+							System.Diagnostics.Process.Start("https://aka.platform.uno/upgrade-uno-packages");
```
Comment (agneszitte) [inline src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs:153]:
```suggestion
							System.Diagnostics.Process.Start(new ProcessStartInfo("https://aka.platform.uno/upgrade-uno-packages") { UseShellExecute = true });
```

**Review by agneszitte — APPROVED** [review]

File: src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs:56]: The stack trace is important here, I'll keep as-is.

File: src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs (hunk 3 above)
Comment (jeromelaban) [reply src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs:153]: `UseShellExecute` is true by default with `System.Diagnostics.Process.Start(string)`

Comment (agneszitte) [reply src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs:154]: Thanks for the info @jeromelaban :)

File: src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs (hunk 1 above)
Comment (agneszitte) [reply src/Uno.UI.RemoteControl.VS/GlobalJsonObserver.cs:56]: oh yeah noted!

**Pull request merged**